src/main/java/com/spotify/confidence/ConfidenceValue.java

```diff
@@ -0,0 +1,547 @@
+    return new List(values.stream().map(ConfidenceValue::of).collect(Collectors.toList()));
+  }
+
+  public static ConfidenceValue.List ofDates(java.util.List<LocalDate> values) {
+    return new List(values.stream().map(ConfidenceValue::of).collect(Collectors.toList()));
+  }
+
+  public static Struct of(Map<String, ConfidenceValue> values) {
+    return new Struct(values);
+  }
+
+  static ConfidenceValue fromProto(com.google.protobuf.Value protoValue) {
+    final com.google.protobuf.Value.KindCase kind = protoValue.getKindCase();
+    switch (kind) {
+      case BOOL_VALUE:
+        return ConfidenceValue.of(protoValue.getBoolValue());
+      case NUMBER_VALUE:
+        return ConfidenceValue.of(protoValue.getNumberValue());
+      case STRING_VALUE:
+        final String stringValue = protoValue.getStringValue();
+        try {
+          return ConfidenceValue.of(Instant.parse(stringValue));
+        } catch (Exception e1) {
+          try {
+            return ConfidenceValue.of(LocalDate.parse(stringValue));
+          } catch (Exception e2) {
+            return ConfidenceValue.of(stringValue);
+          }
+        }
+      case NULL_VALUE:
+        return NULL_VALUE;
+      case STRUCT_VALUE:
+        return Struct.fromProto(protoValue.getStructValue());
+      case LIST_VALUE:
+        final java.util.List<ConfidenceValue> list =
+            protoValue.getListValue().getValuesList().stream()
+                .map(ConfidenceValue::fromProto)
+                .collect(Collectors.toList());
+        return new List(list);
+    }
+    throw new IllegalArgumentException("Unsupported value kind:" + kind);
+  }
+
+  public abstract com.google.protobuf.Value toProto();
+
+  public static class StringValue extends ConfidenceValue {
+    private final String value;
+
+    private StringValue(String value) {
+      this.value = value;
+    }
+
+    @Override
+    public boolean isString() {
+      return true;
+    }
+
+    @Override
+    public String asString() {
+      return value;
+    }
+
+    @Override
+    public com.google.protobuf.Value toProto() {
+      return com.google.protobuf.Value.newBuilder().setStringValue(value).build();
+    }
+
+    @Override
+    public String toString() {
+      return value;
+    }
+  }
+
+  public static class BooleanValue extends ConfidenceValue {
+    private final boolean value;
+
+    private BooleanValue(boolean value) {
+      this.value = value;
+    }
+
+    @Override
+    public boolean isBoolean() {
+      return true;
+    }
+
+    @Override
+    public boolean asBoolean() {
+      return value;
+    }
+
+    @Override
+    public String toString() {
+      return String.valueOf(value);
+    }
+
+    @Override
+    public com.google.protobuf.Value toProto() {
+      return com.google.protobuf.Value.newBuilder().setBoolValue(value).build();
+    }
+  }
+
+  public static class Integer extends ConfidenceValue {
```

Comment:
Why do we need separate types for Integer and Double? Why not just Number? Or ok, does confidence model these as different types? In the proto they're both encoded as double..

Reply:
Let's keep for now.